### ISSUE TITLE
Oracle 19 Create Global Temporary Table Parsing Errors

### ISSUE THREAD

**Issue by sriocreux** [sqlfluff/sqlfluff#4684]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff fails to parse statement to create global temporary table with Oracle dialect

https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/CREATE-TABLE.html#GUID-F9CE0CC3-13AE-4744-A43C-EAC7A71AAAB6

### Expected Behaviour

While linting Oracle SQL from the command line, SQLFluff ought to be able to parse CREATE GLOBAL TEMPORARY TABLE statements using the oracle dialect.

### Observed Behaviour

L: 440 | P:   1 |  PRS | Line 440, Position 1: Found unparsable section: 'CREATE GLOBAL TEMPORARY TABLE HT_AFFAIRE...'

### How to reproduce

Using the `oracle` dialect command line arg, run a SQL statement similar to:

```sql
CREATE GLOBAL TEMPORARY TABLE HT_AFFAIRES (ID CHAR (36 CHAR)) ON COMMIT DELETE ROWS;
```

### Dialect

`oracle`

### Version

sqlfluff version `2.0.3`, Python version `3.7.3`

### Configuration

```
[sqlfluff]
verbose = 0
dialect = oracle
output_line_length = 120
encoding = utf-8
large_file_skip_byte_limit = 0
max_line_length = 120

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 4

[sqlfluff:rules:capitalisation.keywords]
# Keywords
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
